cheatsheets/gleam-for-go-users.md

```diff
@@ -0,0 +1,1460 @@
+
+```go
+import (
+  other "github.com/example/example-go-project/somepkg"
+)
+
+func main() {
+  other.SomeFunc()
+}
+```
+
+#### Gleam
+
+Gleam has as similar feature:
+
+```gleam
+import unix/cat
+import animal/cat as kitty
+// cat and kitty are available
+```
+
+This may be useful to differentiate between multiple modules that would have the same default name when imported.
+
+### Unqualified imports
+
+#### Go
+
+Go does not support unqualified imports
+
+#### Gleam
+
+```gleam
+import animal/cat.{
+  Cat,
+  stroke
+}
+
+pub fn main() {
+  let kitty = Cat(name: "Nubi")
+  stroke(kitty)
+}
+```
+
+Importing common types such as `gleam/order.{Lt, Eq, Gt}` or
+`gleam/option.{Some,None}` can be very helpful.
+
+## Architecture
+
+To iterate a few foundational differences:
+
+1. Programming model: imperative VS functional immutable
+   programming
+2. Guarantees: implicit nillability vs explicit nillability
+3. Runtime model: lightweight coroutines(`goroutines`) VS Erlang/OTP processes
+4. Error handling: error value (and very few panics) VS result type
+5. Distribution: Single static binaries vs Dynamically linked binary that relies on Erlang runtime
+
+### Programming model
+
+- Go is an imperative programming language, with only very limited object-oriented facilities 
+  provided. Gleam offers only functional code style, though it can appear
+  imperative and reads easily thanks to pipes.
+- In Gleam, data structures are never mutated but always updated into new
+  structures. This allows processes that fail to simply restart as there are no
+  mutated objects that can be in an invalid state and take the whole
+  application down (such as in languages like Ruby or Go).
+- Gleam offers syntax to make it easy to extract data out of custom types and
+  update data into new copies of custom types without ever mutating variables.
+  Go sometimes directly mutates references of simple values such as when using
+  pointers or global variables, though this is very much not idiomatic.
+- Gleam allows to rebind variables freely to make it easy to update data
+  structures by making a copy and binding it to the existing variable.
+- Go features a massive, powerful standard library centered around simple interfaces.
+  Everything from string manipulation to the tools to create an http server are included
+  out of the box.  It has been tuned for over 15 years for performance and has strong
+  guarantees about backwards and forwards compatibility.
+- Gleam allows you to opt into a smaller, well polished and consistent standard
+  library.
+
+### Guarantees and types
+
+- Both Go and Gleam features strong static typing.
+- Both Go and Gleam values are not automatically cast.
+- Both Go and Gleam's comparison operators are very strict and limited, any other
+  comparisons and conversions must happen via function calls or other explicit type
+  conversion mechanisms.
+- Go's checks happen at compile time (except for reflection), Gleam uses the compiler
```

Comment:
This sounds a bit like Gleam does runtime checks while Go do more compile time checks?